### PR TITLE
Adding line and multiple point output

### PR DESCRIPTION
**Pull request by mpentek**

### CODE REVIEW

File: kratos/python_scripts/multiple_points_output_process.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            raise Exception('No positions were specified"')
+        if positions.Size2() != 3:
+            raise Exception('The positions have to be provided with 3 coordinates!')
+
+        self.point_output_processes = []
+        params.RemoveValue("positions")
+        params.AddEmptyValue("position")
+        position_vec = KratosMultiphysics.Vector(3)
+      
+        if params["output_file_name"].GetString().endswith(".dat"):
```
Comment (philbucher) [inline kratos/python_scripts/multiple_points_output_process.py]: is this needed?
I think the `PointOutputProcess` takes care of that

File: kratos/python_scripts/line_output_process.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                positions[k,2] = current_position[2]
+
+        params.RemoveValue("start_point")
+        params.RemoveValue("end_point")
+        params.RemoveValue("sampling_points")
+
+        params.AddEmptyValue("positions")
+        params["positions"].SetMatrix(positions)
+
+        if params["output_file_name"].GetString().endswith(".dat"):
```
Comment (philbucher) [inline kratos/python_scripts/line_output_process.py]: is this needed?
I think the `PointOutputProcess` takes care of that

**Review by philbucher — COMMENTED** [review]
Nice and small PR :+1: 

Since I was involved a lot in this I leave approval to @msandre

File: kratos/python_scripts/multiple_points_output_process.py (hunk 1 above)
Comment (mpentek) [reply kratos/python_scripts/multiple_points_output_process.py]: In the multiple_points_output_process.py the check in line 50 is needed, as it adds an index, which would be bad if it were to be added after an extension ".dat".

File: kratos/python_scripts/line_output_process.py (hunk 2 above)
Comment (mpentek) [reply kratos/python_scripts/line_output_process.py]: In the line_output_process the check in line 85 is not necessary, I will remove it. See the previous comment.

File: kratos/python_scripts/multiple_points_output_process.py (hunk 1 above)
Comment (philbucher) [reply kratos/python_scripts/multiple_points_output_process.py]: ok

File: kratos/python_scripts/line_output_process.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return LineOutputProcess(Model, settings["Parameters"])
+
+class LineOutputProcess(KratosMultiphysics.Process):
+    """This process writes output for several points along a line to a file
+    Internally it holds an object of type "MultiplePointsOutputProcess"
+    Usage:
+        - specifying start and end point defining the line
+        - specifying the number of sampling points alond the line (start and end points will be included)
```
Comment (msandre) [inline kratos/python_scripts/line_output_process.py]: typo

**Review by msandre — DISMISSED** [review]
Thanks!

**Review by msandre — DISMISSED** [review]

**Comment by mpentek** [comment]
@philbucher @msandre I get some "Not up to standards" by code review Codacy/PR Review. However, I am using the syntax as it has been done up until now.

Such as:

Using type() instead of isinstance() for a typecheck. ==>> 
if(type(settings) != KratosMultiphysics.Parameters):

Any ideas/hints on how to proceed? It is requesting to use isinastance() instad of type() syntax.

**Comment by philbucher** [comment]
The ones related to the type checking can be ignored

**Comment by msandre** [comment]
@mpentek I think you can use either.

**Comment by mpentek** [comment]
It does not want to let me. What confuses me is that in other files throughout the Kratos project is is doing the check with type(). Is there some new change with respect to code quality standards?

@philbucher how do I ignore it (or make GitHub/Codacy ignore it?)

**Comment by philbucher** [comment]
@mpentek codacy is not mandatory for merging, only travis
=> i.e. you can merge even if it complains
It only suggests things, the developer has to decide if it is applicable

regarding the typechecking as mike says in this case it doesn't matter, but I would use `type` since it is used everywhere in Kratos

(btw I don't know why it suggests isinstance over type, I think this has to be chosen depending on the situation => try to compare bool and ints in python with isinstance 
Spoiler alert: bool derives from int which is really nice for debugging ;) )

**Comment by mpentek** [comment]
@philbucher ok, so I will disregard the warnings. I still need a valid approving review seemingly.

**Review by philbucher — APPROVED** [review]
Yep the remaining warnings can be safely ignored

**Pull request merged**